tools/code-generation/smithy/codegen/cpp-smoke-tests-codegen/src/main/java/com/amazonaws/util/awsclientsmithygenerator/generators/SmokeTestsParser.java

```diff
@@ -0,0 +1,348 @@
+
+        model.getOperationShapes().stream().
+        filter(operationShape -> operationShape.getInput().isPresent()).
+        filter(operationShape -> operationShape.getTrait(SmokeTestsTrait.class).isPresent() ).
+        filter(operationShape ->  operationToServiceMap.containsKey(operationShape.getId()) ).
+        forEach(operationShape -> {
+            SmokeTestsTrait smokeTestsTrait = operationShape.getTrait(SmokeTestsTrait.class).get();
+            //get serviceShape
+            String serviceName = operationToServiceMap.get(operationShape.getId());
+            ServiceShape serviceShape = serviceShapeMap.get(serviceName);
+            //for one service, we have one smoke test file
+            //So, we for a given service we import dependencies (symbol types and their respective headers)
+            List<SmokeTestData> tests = serviceSmokeTestsMap.getOrDefault(serviceShape, new ArrayList<>());
+
+            tests.addAll(parseSmokeTests(
+                smokeTestsTrait, 
+                operationShape,
+                serviceShape ));
+            
+            serviceSmokeTestsMap.put(serviceShape, tests);
+
+        });
+        
+        return serviceSmokeTestsMap;
+    }
+
+    @Override
+    public void run(){
+
+        Map<ServiceShape, List<SmokeTestData> > smoketests =  extractServiceSmokeTests();
+
+        //make service specific folder
+        smoketests.forEach((serviceShape, value) -> {
+            try {
+                SmokeTestsSourceDelegator delegator = new SmokeTestsSourceDelegator(this.context.getFileManifest(), this.symbolProvider, value);
+                SmokeTestsCMakeDelegator cmakedelegator = new SmokeTestsCMakeDelegator(this.context.getFileManifest(), this.symbolProvider);
+
+                String client = getServiceName(serviceShape);
+                String c2jClientname = SmithyC2JNamespaceMap.getInstance().getC2JServiceName(toKebabCase(client));
+
+                Path relativePath = Paths.get(c2jClientname);
+                System.out.println(toKebabCase(client) + " mapped to " + c2jClientname);
+
+                delegator.useFileWriter(relativePath.toString() + "/" + removeSpaces(client) + "SmokeTests.cpp", client, writer -> {
+                    System.out.printf("generating smoke test source code=%s%n", relativePath.toString() + "/" + removeSpaces(client) + "SmokeTests.cpp");
+                    writer.generate();
+                });
+
+                cmakedelegator.useFileWriter(relativePath.toString() + "/" + "CMakeLists.txt", c2jClientname, writer -> {
+                    System.out.printf("generating smoke test cmake code=%s%n", relativePath.toString() + "/" + "CMakeLists.txt");
+                    writer.generate();
+                });
+
+                delegator.flushWriters();
+                cmakedelegator.flushWriters();
+            } catch (Exception e) {
+                System.err.println("Exception detected=" + e.toString());
```

Comment:
dont catch the exception, let the exception bubble up as a runtime, and just let it fail if a exception is thrown

Reply:
ok, removed. The only thing is that first exception will bubble up to exit

Reply:
thats fine thats what we want